hq/app/api/Snyk.scala

```diff
@@ -2,40 +2,36 @@ package api
 
 import model._
 import play.api.libs.json._
-import play.api.libs.ws.{WSClient, WSResponse}
+import play.api.libs.ws.{WSClient, WSRequest, WSResponse}
 import utils.attempt.{Attempt, FailedAttempt, Failure}
 
 import scala.concurrent.ExecutionContext
 import scala.util.control.NonFatal
 
 object Snyk {
 
-  def getSnykOrganisations(token: SnykToken, wsClient: WSClient)(implicit ec:ExecutionContext): Attempt[WSResponse] = {
+  def snykRequest(token:SnykToken, url:String, wsClient: WSClient) = wsClient.url(url).addHttpHeaders("Authorization" -> s"token ${token.value}")
+
+  def getSnykOrganisations(token: SnykToken, wsClient: WSClient)(implicit ec: ExecutionContext): Attempt[WSResponse] = {
     val snykOrgUrl = "https://snyk.io/api/v1/orgs"
-    val futureResponse = wsClient.url(snykOrgUrl)
-      .addHttpHeaders("Authorization" -> s"token ${token.value}")
-      .get
-    Attempt.fromFuture(futureResponse) { case NonFatal(e) =>
-      val failure = Failure(e.getMessage, "Could not read organisations from Snyk", 502, None, Some(e))
-      FailedAttempt(failure)
-    }
+    val futureResponse = snykRequest(token, snykOrgUrl, wsClient).get
+    handleFuture(futureResponse, "organisation")
   }
 
-  def getProjects(token: SnykToken, id: String, wsClient: WSClient)(implicit ec:ExecutionContext): Attempt[WSResponse] = {
-    val snykProjectsUrl = s"https://snyk.io/api/v1/org/$id/projects"
-    val a = wsClient.url(snykProjectsUrl)
-        .addHttpHeaders("Authorization" -> s"token ${token.value}")
-        .get()
-    Attempt.fromFuture(a) { case NonFatal(e) =>
-      val failure = Failure(e.getMessage, "Could not read projects from Snyk", 502, None, Some(e))
-      FailedAttempt(failure)
+  def getProjects(token: SnykToken, organisations: List[SnykOrganisation], wsClient: WSClient)(implicit ec: ExecutionContext): Attempt[List[(SnykOrganisation, String)]] = {
+    Attempt.traverse(organisations) { organisation =>
+      val snykProjectsUrl = s"https://snyk.io/api/v1/org/${organisation.id}/projects"
+      val futureResponse = snykRequest(token, snykProjectsUrl, wsClient).get
+        .transform(response => (organisation, response.body), f => f )
```

Comment:
`transform` is a method in Future for mapping the value in the Future and providing a handler for errors in one go. Since you are using identity for the error handling this would be better written as `map`. At that point though, I would suggest it's better to map the resulting Attempt rather than the intermediate Future. It would reduce the dissonance of mapping over different effects in different places.

This is also a step towards splitting the logic of processing the WSResponse from the code that performs the request. A common pattern for the AWS calls is the following (simplified for our purposes here).

```scala
def doSomethingWithAResponse(response: ServiceResponse): MyType = ???

def awsCall(arg: Something, client: ServiceClient): MyType = {
  makeAnAttempt(client.call(arg)).map(doSomethingWithAResponse)
}
```
This means we have a function that fetches the response separate to a function that does things with that response. I don't mean to say you have to do exactly this here (they are different!) but it'd be great to see minimal logic/code within a traverse call, in particular.